server/app/controllers/ErrorHandler.java

```diff
@@ -100,4 +109,10 @@ static Optional<Throwable> findThrowableByTypes(
     }
     return Optional.empty();
   }
+
+  @Override
+  public CompletionStage<Result> onNotFound(RequestHeader request, String message) {
+    return CompletableFuture.completedFuture(
+        Results.ok(notFoundPage.render(request, messagesApi.preferred(request))));
```

Comment:
suggestion (blocking): the response code for not found should be 404, please change to `Results.notFound(...)`